### PR TITLE
Update dependencies and bump version

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -11,15 +11,18 @@ repository cardano-haskell-packages
...
--- Bump this if you need newer packages from CHaP
-index-state: cardano-haskell-packages 2022-11-11T00:00:00Z
+index-state:
+  , hackage.haskell.org 2023-01-30T23:55:53Z
+  , cardano-haskell-packages 2023-02-14T10:24:01Z
 
 packages: ./.
 
+allow-newer:
+    monoidal-synchronisation:base
```
Comment (michaelpj) [inline cabal.project:24]: This is potentially a problem. Is there an upstream issue?

Comment (michaelpj) [reply cabal.project:24]: It's in ouroboros-network, so should they not have released a compatible one already?

Comment (erikd) [reply cabal.project:24]: Yes, there is an upstream issue.

Comment (erikd) [reply cabal.project:24]: Yes, they should have fixed this.

File: cabal.project [hunk 2]
```diff
@@ -11,15 +11,18 @@ repository cardano-haskell-packages
     c00aae8461a256275598500ea0e187588c35a5d5d7454fb57eac18d9edb86a56
     d4a35cd3121aa00d18544bb0ac01c3e1691d618f462c46129271bccf39f7e8ee
 
--- See ouroboros-network/CONTRIBUTING for some Nix commands you will need 
+-- See ouroboros-network/CONTRIBUTING for some Nix commands you will need
 -- to run if you update either of these.
 -- Bump this if you need newer packages from Hackage
-index-state: 2022-10-31T00:00:00Z
```
Comment (angerman) [inline cabal.project:17]: This needs to be duplicated due to some tooling defects. Should look similar to this:

https://github.com/input-output-hk/cardano-node/blob/c5ca9c2884a65737a8eec7afaac95ca2e44da47d/cabal.project#L15-L19

**Review by angerman — CHANGES_REQUESTED** [review]

File: cabal.project (hunk 1 above)
Comment (coot) [reply cabal.project:24]: It's fixed now: https://github.com/input-output-hk/cardano-haskell-packages/pull/132

Comment (erikd) [reply cabal.project]: Upstream fixed, CHaP `index-state` updated, `allow-newer` removed.

File: cabal.project (hunk 2 above)
Comment (erikd) [reply cabal.project:17]: Its now:
```
index-state:
  , hackage.haskell.org 2023-01-30T23:55:53Z
  , cardano-haskell-packages 2023-02-16T13:42:30Z
```

Comment (angerman) [reply cabal.project:17]: Yes? But it needs the reduplifafion of the index state :-/ 

```

index-state: 2023-01-20T05:50:56Z

index-state:
  , hackage.haskell.org 2023-01-20T05:50:56Z
  , cardano-haskell-packages 2022-12-14T00:40:1
```

Index-state needs to be listed twice. Once with repos, and once without for the hackage index state.

Yes this is stupid and our tools are defective. But it's how things currently stand :-(

**Comment by erikd** [comment]
> Need another index-state line m(

But I was told that was wrong!

**Review by angerman — APPROVED** [review]
LGTM

**Pull request merged**